src/inference/AdvancedSMC.jl

```diff
@@ -138,46 +132,75 @@ function AbstractMCMC.step!(
     # check that we received a real iteration number
     @assert iteration >= 1 "step! needs to be called with an 'iteration' keyword argument."
 
-    # grab the weights
+    # grab the weight
     pc = spl.state.particles
-    Ws = getweights(pc)
+    weight = getweight(pc, iteration)
 
     # update the master vi
     particle = pc.vals[iteration]
     params = tonamedtuple(particle.vi)
+
+    # This is pretty useless since we reset the log probability continuously in the
+    # particle sweep.
```

Comment:
thanks for the note, also see my comment above about assume and observe functions.